File: rust/src/models/endpoint_deleted_event.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::endpoint_deleted_event_data::EndpointDeletedEventData;

/// Sent when an endpoint is deleted.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct EndpointDeletedEvent {
    pub data: EndpointDeletedEventData,

    pub r#type: String,
}

impl EndpointDeletedEvent {
    pub fn new(data: EndpointDeletedEventData, r#type: String) -> Self {
        Self { data, r#type }
    }
}

File: javascript/src/models/endpointDeletedEvent.ts
// this file is @generated
import {
  type EndpointDeletedEventData,
  EndpointDeletedEventDataSerializer,
} from "./endpointDeletedEventData";

/** Sent when an endpoint is deleted. */
export interface EndpointDeletedEvent {
  data: EndpointDeletedEventData;
  type: string;
}

export const EndpointDeletedEventSerializer = {
  _fromJsonObject(object: any): EndpointDeletedEvent {
    return {
      data: EndpointDeletedEventDataSerializer._fromJsonObject(object["data"]),
      type: object["type"],
    };
  },

  _toJsonObject(self: EndpointDeletedEvent): any {
    return {
      data: EndpointDeletedEventDataSerializer._toJsonObject(self.data),
      type: self.type,
    };
  },
};
